articles/references/module_builder.md

```diff
@@ -0,0 +1,81 @@
+---
+uid: ModuleBuilder
+---
+# Module Builder
+
+The Module Builder is one of the powerful features of Intent Architect which allows a developer to craft their own code-generation module. This will allow a developer to generate custom defined code at the click of a button once that module is installed (while its being Intent Architect Managed). Think of the power!
+
+Generally to write your own Module required that you had to put up a lot of code to create a Template and the like, but the Module Builder takes care of that by modeling a Module the same way you would model your own application. Think of all the other possibilities of using Intent Architect!
+
+Please visit the Tutorial section to see how to work with the [Module Builder](xref:TutorialCreateModule).
+
+>[!NOTE]
+>The scope of this article is to explain the concepts found in the Module Builder. If you wish to find out more about Modules or Templates themselves, please navigate to the relevant section on the panel to the left.
+
+## Breakdown
+
+![Module Builder Components](images/ModuleBuilderComponentBreakdown.png)
+_Module Builder Components_
+
+For the purposes of this document, we will only stick to the following components below. For information on the rest of the components outlined in the image above, please visit the [Intent Architect Concepts](xref:IntentArchitectConcepts) page.
+
+### File Template
+Creates a Template class that will allow you to generate a text file for any type of text content (including any programming language).
+
+### C# Template
+Since Intent Architect supports intelligent weaving in C# files, it will create a template that will set this up for you. It is recommended to use this type of template for all C# classes as the [weaving](xref:RoslynWeaver) systems allow user-managed code to co-exist with Intent-managed code within the same file.
+
+### Template Decorator
+Creates a Decorator class that you can use to add content to an existing template by hooking into its extensibility points. Think of this as a Object Oriented Class that extends an Abstract class and has to implement all the virtual / abstract methods but in a Template sense.
+
+### Modeler
+If you need to fetch meta-data from a modeler that is other than `Domain` or `Services`, you need to add it here with the appropriate name so that your template can reference it. See [below](#modeler).
```

Comment:
```suggestion
Creates an element which represents a "reference" to a modeler. These references are needed if you are going to use meta-data from a custom modeler that is other than the `Domain` or `Services` modeler. See [below](#modeler).
```

Question, if one puts `[](#modeler)` does it pick up the heading name as the link title? If so then we could rather do:
```suggestion
Creates an element which represents a "reference" to a modeler. These references are needed if you are going to use meta-data from a custom modeler that is other than the `Domain` or `Services` modeler. See [](#modeler) below.
```

We should have a modeler article somewhere, if we don't already, then lets make an issue to get one made.